### PR TITLE
Configure to work with heroku

### LINKED ISSUE #11: INFRA: create a heroku config file so this can be deployed on heroku

### PR DESCRIPTION
**Pull request by Kartikdot**
Fixes #11 
Changes:
1. Added Procfile.
2. Modified connect.ts to createConnection with URL in production environment.
3. Modified package.json to so that prestart script only runs with npm run start:dev and not in production as 'tsc' is not recognized (typescript is a dev dependency).
4. changed run.ts to take PORT from environment variables if available.

Additional fixes:
1. Length comparison for custom shortcode was wrong
2. used the already implemented createRandomShortCode method.

### CODE REVIEW

File: src/controllers/shortcode.controller.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export async function createSpecificShortCode(shortCode: string, longUrl: string
     const newEntity = new ShortCode()
 
     // check if shortCode is less than 8 chars in length 
-    if(shortCode.length < 8) {
```
Comment (Kartikdot) [inline src/controllers/shortcode.controller.ts:21]: was this generated by copilot?

File: src/db/connect.ts [hunk 2]
```diff
@@ -1,14 +1,28 @@
...
+    if (process.env.NODE_ENV == 'production') {
+        options = {
+            type: "postgres",
+            url: process.env.DATABASE_URL,
+            logging: "all",
+            logger: "advanced-console",
+            entities: [ShortCode],
+            synchronize: true
+        };
+    };
```
Comment (championswimmer) [inline src/db/connect.ts:25]: Codacy has a fix for the issue: [Unnecessary semicolon.](https://app.codacy.com/gh/scaleracademy/sclr.ac-backend/pullRequest?prid=7702321)
```suggestion
    }
```

File: src/db/connect.ts [hunk 3]
```diff
@@ -1,14 +1,28 @@
...
+        username: "sclrac",
+        password: "sclrac",
+        database: "sclrac",
+        logging: "all",
+        logger: "advanced-console",
+        entities: [ShortCode],
+        synchronize: true,
+    };
+    
+    if (process.env.NODE_ENV == 'production') {
```
Comment (championswimmer) [inline src/db/connect.ts:16]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/scaleracademy/sclr.ac-backend/pullRequest?prid=7702321)
```suggestion
    if (process.env.NODE_ENV == "production") {
```

File: src/controllers/shortcode.controller.ts (hunk 1 above)
Comment (championswimmer) [reply src/controllers/shortcode.controller.ts:21]: haha, yes it was :P pitfalls for AI

**Review by championswimmer — APPROVED** [review]

File: src/controllers/shortcode.controller.ts (hunk 1 above)
Comment (Kartikdot) [reply src/controllers/shortcode.controller.ts:21]: haha nice.

**Pull request merged**